[BUG] There is already a Construct with name 'userPoolClientSecretArn*-up-client' in Stack ..

**Describe the bug**
When the secret parameters from the CognitoStack is retrieved there's always an error such as:
```
Caused by: software.amazon.jsii.JsiiException: There is already a Construct with name 'userPoolClientSecretArn<some_name_here>-up-client' in Stack [<Stack_name_here>]
```
I suspect the ids for the string parameters to retrieve the secret arn and the secret itself are the same, I mean same id for both construct

**To Reproduce**
Steps to reproduce the behavior:
1. Deploy cognito stack
2. Deploy another stack which uses it's output params
3. See error

**Expected behavior**
To deploy services which rely on the cognito ouput params to deploy properly

--- a/src/main/java/org/wcdevs/blog/cdk/CognitoStack.java
+++ b/src/main/java/org/wcdevs/blog/cdk/CognitoStack.java
@@ -107,13 +107,6 @@ public final class CognitoStack extends Stack {
     return clientParams.stream().map(clientParam -> userPoolClient(scope, userPool, clientParam));
   }
 
-  public static ISecret getUserPoolClientSecret(Stack scope, ApplicationEnvironment appEnv) {
-    var clientName = clientName(appEnv.getApplicationName());
-    var arnParamHolder = clientSecretArnParamHolder(clientName);
-    var arn = getParameterUserPoolClientSecretArn(scope, appEnv);
-    return Secret.fromSecretCompleteArn(scope, arnParamHolder, arn);
-  }
-
   private static String clientSecretArnParamHolder(String userPoolClientName) {
     return PARAM_USER_POOL_CLIENT_SECRET_ARN + userPoolClientName;
   }
@@ -208,10 +201,14 @@ public final class CognitoStack extends Stack {
   }
 
   private static void createStringParameter(Stack scope, String envName, String id, String value) {
-    StringParameter.Builder.create(scope, id)
+    StringParameter.Builder.create(scope, parameterId(id))
                            .parameterName(createParameterName(envName, id))
                            .stringValue(value)
                            .build();
+  }
+
+  private static String parameterId(String id) {
+    return joinedString(DASH_JOINER, CONSTRUCT_NAME, "param", id);
   }
 
   private static String createParameterName(String envName, String parameterName) {
@@ -282,6 +279,12 @@ public final class CognitoStack extends Stack {
   // endregion
 
   // region get output params
+  public static ISecret getUserPoolClientSecret(Stack scope, ApplicationEnvironment appEnv) {
+    var clientName = clientName(appEnv.getApplicationName());
+    var secConstructId = joinedString(DASH_JOINER, clientSecretArnParamHolder(clientName), "sec");
+    var arn = getParameterUserPoolClientSecretArn(scope, appEnv);
+    return Secret.fromSecretCompleteArn(scope, secConstructId, arn);
+  }
 
   /**
    * Returns the output parameters from this stack creation except the user pool clients secret
@@ -303,7 +306,8 @@ public final class CognitoStack extends Stack {
   }
 
   public static String getParameter(Stack scope, String appEnv, String id) {
-    return StringParameter.fromStringParameterName(scope, id, createParameterName(appEnv, id))
+    return StringParameter.fromStringParameterName(scope, parameterId(id),
+                                                   createParameterName(appEnv, id))
                           .getStringValue();
   }
 

--- a/src/test/java/org/wcdevs/blog/cdk/CognitoStackTest.java
+++ b/src/test/java/org/wcdevs/blog/cdk/CognitoStackTest.java
@@ -176,10 +176,6 @@ class CognitoStackTest {
     var envName = randomString();
     var appName = randomString();
     var secretMock = mock(ISecret.class);
-
-    var clientName = Util.joinedString(Util.DASH_JOINER, appName, "up", "client");
-    var arnParamHolder = CognitoStack.PARAM_USER_POOL_CLIENT_SECRET_ARN
-                         + clientName;
     var arn = randomString();
 
     try (
@@ -190,7 +186,7 @@ class CognitoStackTest {
       when(stringParam.getStringValue()).thenReturn(arn);
       mockedStringParameter.when(() -> StringParameter.fromStringParameterName(any(), any(), any()))
                            .thenReturn(stringParam);
-      mockedSecret.when(() -> Secret.fromSecretCompleteArn(any(), eq(arnParamHolder), eq(arn)))
+      mockedSecret.when(() -> Secret.fromSecretCompleteArn(any(), any(), any()))
                   .thenReturn(secretMock);
       var appEnv = mock(ApplicationEnvironment.class);
       when(appEnv.getEnvironmentName()).thenReturn(envName);

